### PR TITLE
[IMP][9.0] Partner names order configurable

### PR DESCRIPTION
**Pull request by lasley**
This is a forward port of `partner_firstname` changes from #227. Basically just a cherry-pick and a few style adjustments as noted below:

* Prefer odoo to openerp & remove data tags
* Move exceptions file out of models

### CODE REVIEW

File: partner_firstname/README.rst
Comment (pedrobaeza) [inline partner_firstname/README.rst]: s/8.0/9.0

File: partner_firstname/README.rst
Comment (pedrobaeza) [inline partner_firstname/README.rst]: Remove this part

**Review by pedrobaeza — APPROVED** [review]
Minor changes, but overall :+1: 

Well, when Travis goes green

File: partner_firstname/views/base_config_view.xml
Comment (lasley) [inline partner_firstname/views/base_config_view.xml]: No string selector in >8

Comment (pedrobaeza) [reply partner_firstname/views/base_config_view.xml]: Yeah, that's it. Let's hope there's a new shining name attribute around...

**Comment by lasley** [comment]
Pesky rebase got me on this one - missed an import! Alright we're good now & Pedro's comments attended to, I'll squash up before we merge. 

I'll be submitting the v10 in a bit as well, that's the version I really need it on 😉

File: partner_firstname/views/base_config_view.xml
Comment (lasley) [reply partner_firstname/views/base_config_view.xml]: There was! `email_label` 😄

**Comment by pedrobaeza** [comment]
Thanks for taking care of this version.

**Comment by lasley** [comment]
Added icon fix from https://github.com/OCA/partner-contact/pull/334#issuecomment-275819078

**Comment by whnr** [comment]
Do I understand correctly that this is not available in the App? How can I use it then? Every search for this topic points here but it seems to be not available in the released version in Odoo.

**Comment by lasley** [comment]
@floweh This is not available in the v10 version until someone else adds an accepted review, which will allow for a merge

**Comment by whnr** [comment]
@lasley yes, but this is the V9 I need… The changes seem to be approved.

**Comment by lasley** [comment]
Sorry I meant v9, which is the version of this PR. 

> The changes seem to be approved.

A change is not approved without two reviews, [per the guidelines](https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#review). It would be amazing if you went ahead and provided one, so that we could look at merging this.

**Comment by pedrobaeza** [comment]
As this is has been long outstanding and it has my bless, we can merge it and remove pending tasks.

**Pull request merged**
